src/Raven.Server/Documents/Queries/GraphQueryRunner.cs

```diff
@@ -164,9 +180,8 @@ private async Task<(List<Match> Matches, GraphQueryPlan QueryPlan)> GetQueryResu
                             matchResults[i] = default;
                     }
                 }
-            }
-
-            return (matchResults, qp);
+            }            
+            return (matchResults.Skip(query.Start).Take(query.PageSize).ToList(), qp);
```

Comment:
The output is a list so I need to generate the result, do you want me to change the result to be IEnumerable (this will have a ripple affect on the code)? 
I could save the two IEnumerable calls using a while loop.

Reply:
If `query.Start` is 0, then you don't need to allocate anything.
Simply set remove anything from the end of the list until it hit `PageSize`